### PR TITLE
Cblas and lapacke includes

### PR DESCRIPTION
**Pull request by jthies**
C users may want to know where to find the cblas.h and lapacke.h headers, so I added the headers property to two blas/lapack providers (Intel-mkl and netlib-blas), as suggested by @davydden. Also adjusted the recently added ghost package to make use of this.

### CODE REVIEW

File: var/spack/repos/builtin/packages/ghost/package.py [hunk 1]
```diff
@@ -78,9 +75,12 @@ def cmake_args(self):
                 '-DGHOST_USE_ZOLTAN:BOOL=%s'
                 % ('ON' if '+zoltan' in spec else 'OFF'),
                 '-DBUILD_SHARED_LIBS:BOOL=%s'
-                % ('ON' if '+shared' in spec else 'OFF'), cblas_include_dir
+                % ('ON' if '+shared' in spec else 'OFF'),
+                '-DCBLAS_INCLUDE_DIR:STRING=%s'
+                % format(spec['blas:c'].headers.directories[0]),
```
Comment (davydden) [inline var/spack/repos/builtin/packages/ghost/package.py]: since you did not use any query parameters, then there is no need for `:c` here.

**Comment by davydden** [comment]
@alalazo ping.

**Comment by davydden** [comment]
@alalazo ping ping.

**Comment by adamjstewart** [comment]
Superseded by #7501.

**Pull request closed without merging**